libafl/src/state/mod.rs

```diff
@@ -262,22 +302,72 @@ pub struct StdState<I, C, R, SC> {
     /// The current index of the corpus; used to record for resumable fuzzing.
     corpus_id: Option<CorpusId>,
     stage_stack: StageStack,
+    #[cfg(feature = "dump_state")]
+    dump_state_dir: Option<PathBuf>,
     phantom: PhantomData<I>,
 }
 
+/// The standard state dump
+#[cfg(all(feature = "std", feature = "dump_state"))]
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(bound = "I: serde::de::DeserializeOwned")]
+pub struct StdStateDump<I>
+where
+    I: Input,
+{
+    /// Fuzzer start time.
+    pub start_time: Duration,
+
+    /// Total nb of execs
+    pub executions: u64,
+
+    /// Loaded inputs
+    pub testcases: Vec<TestcaseDump<I>>,
+}
+
 impl<I, C, R, SC> UsesInput for StdState<I, C, R, SC>
 where
     I: Input,
 {
     type Input = I;
 }
 
+#[cfg(all(feature = "std", feature = "dump_state"))]
+impl<I, C, R, SC> MaybeCanDumpState for StdState<I, C, R, SC>
+where
+    C: Corpus<Input = I>,
+    I: Input,
+{
+    type StateDump = StdStateDump<I>;
+
+    fn dump_state_dir(&self) -> Option<&PathBuf> {
+        self.dump_state_dir.as_ref()
+    }
+
+    fn gen_dump_state(&mut self) -> Result<Self::StateDump, Error> {
+        let mut tcs: Vec<TestcaseDump<I>> = Vec::new();
+        for corpus_id in self.corpus.ids() {
+            let mut tc = self.corpus.get(corpus_id)?.clone();
+            let tc_ref = tc.get_mut();
+            tc_ref.load_input(&self.corpus)?;
+            tcs.push(tc_ref.clone().try_into()?);
+        }
+
+        Ok(StdStateDump {
```

Comment:
Why not the dump the whole state now?

Reply:
at this point we are doing this just for our experiment

Reply:
but to answer the question, 
we wanted to parse this in a standalone tool where modules such as `Scheduler` does not exist, but this cause a problem because we have to add types to it to compile our tool.

That's why we just select a few members from it and make it into another struct (but again, it's just for  the experiment

Reply:
You can dump to JSON and remove the keys you don't want
or maybe you can change the deserializer to ignore unneeded types